### PR TITLE
Introduce `TaskLike`

### PR DESCRIPTION
**Pull request by pjurczenko**
Solves #20, #36.

### CODE REVIEW

File: core/shared/src/main/scala/monix/bio/TaskLike.scala [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+      def apply[A](thunk: () => A): Task[A] =
+        BIO.Eval(thunk)
+    }
+
+  /**
+    * Converts `scala.util.Either` into a `Task`.
+    */
+  implicit def fromEither[E <: Throwable]: TaskLike[Either[E, *]] =
+    new TaskLike[Either[E, *]] {
+      def apply[A](fa: Either[E, A]): Task[A] =
```
Comment (Avasil) [inline core/shared/src/main/scala/monix/bio/TaskLike.scala]: Would be nice to be able to support `BIO[E, A]` here but I'll leave it for the future

File: core/jvm/src/test/scala/monix/bio/TaskLikeConversionsJava8Suite.scala [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    s.tick()
+    assertEquals(f.value, None)
+
+    f.cancel()
+    s.tick()
+    assertEquals(f.value, None)
+
+    assert(future.isDone)
+    assert(future.isCancelled)
+    s.tick()
```
Comment (Avasil) [inline core/jvm/src/test/scala/monix/bio/TaskLikeConversionsJava8Suite.scala]: I think we can remove this last `tick` and `assert`

Comment (Avasil) [inline core/jvm/src/test/scala/monix/bio/TaskLikeConversionsJava8Suite.scala]:
```suggestion
```

File: core/jvm/src/test/scala/monix/bio/TaskLikeConversionsJava8Suite.scala [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    assertEquals(f.value, None)
```
Comment (Avasil) [inline core/jvm/src/test/scala/monix/bio/TaskLikeConversionsJava8Suite.scala]:
```suggestion
```

File: core/shared/src/test/scala/monix/bio/TaskLikeConversionsSuite.scala [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+    val source = CIO(IO { executed = true; throw ex })
+    val task = BIO.from(source)
+
+    assert(!executed)
+    val f = task.runToFuture
+    assert(executed)
+    assertEquals(f.value, Some(Success(Left(ex))))
+  }
+
+  test("BIO.from converts Future0") { implicit s =>
```
Comment (Avasil) [inline core/shared/src/test/scala/monix/bio/TaskLikeConversionsSuite.scala]:
```suggestion
  test("BIO.from converts Function0") { implicit s =>
```

**Comment by pjurczenko** [comment]
@Avasil thanks for the review, I have updated the pull request.

**Pull request merged**